keras/engine/topology.py

```diff
@@ -2465,11 +2469,14 @@ def load_weights_from_hdf5_group(self, f):
             # we batch weight value assignments in a single backend call
             # which provides a speedup in TensorFlow.
             weight_value_tuples = []
+            backend = f.attrs.get('keras_backend')
             for k, name in enumerate(layer_names):
                 g = f[name]
                 weight_names = [n.decode('utf8') for n in g.attrs['weight_names']]
                 weight_values = [g[weight_name] for weight_name in weight_names]
                 layer = flattened_layers[k]
+                if backend and backend != K._BACKEND:
+                    weight_values = layer.convert_weights(weight_values, backend)
```

Comment:
It is be preferable to confine changes to this loop instead of introducing a new layer method. Here, we just need a loop that builds the `weight_values` list and flips convolution kernels if `layer` is an instance of a convolution layer.